src/postgres/tree/PostgresFunctionsTreeItem.ts

```diff
@@ -32,13 +36,59 @@ export class PostgresFunctionsTreeItem extends AzureParentTreeItem<ISubscription
     }
 
     public async loadMoreChildrenImpl(): Promise<PostgresFunctionTreeItem[]> {
+        const rows: IPostgresProceduresQueryRow[] = await this.listFunctions();
+        this._functionsAndSchemas = {};
+        for (const row of rows) {
+            if (this._functionsAndSchemas[row.name]) {
+                this._functionsAndSchemas[row.name].push(row.schema);
+            } else {
+                this._functionsAndSchemas[row.name] = [row.schema];
+            }
+        }
+
+        return rows.map(row => new PostgresFunctionTreeItem(
+            this,
+            row,
+            this._functionsAndSchemas[row.name].length > 1
+        ));
+    }
+
+    public async createChildImpl(_context: IActionContext): Promise<PostgresFunctionTreeItem> {
+        const schemas: string[] = await this.listSchemas();
+        let schema: string;
+
+        if (schemas.length === 1) {
+            schema = schemas[0];
+        } else {
+            const schemaQuickPicks = schemas.map(s => { return { label: s }; });
+            schema = (await ext.ui.showQuickPick(schemaQuickPicks, {
+                placeHolder: localize('selectSchema', 'Select schema for new function...')
+            })).label;
+        }
+
+        const name: string = (await ext.ui.showInputBox({
+            prompt: localize('enterFunctionName', 'Enter function name'),
+            validateInput: value => this.validateFunctionName(value, schema)
+        })).trim();
+
+        const schemasContainingFunction: string[] = this._functionsAndSchemas[name] || [];
+        const isDuplicate: boolean = schemasContainingFunction.length > 1 || (!schemasContainingFunction.includes(schema) && schemasContainingFunction.length === 1);
```

Comment:
What edge case is this? `(!schemasContainingFunction.includes(schema) && schemasContainingFunction.length === 1);`

Can you have two functions with the same name and schema?

Reply:
This was supposed to handle when one other schema had this function name already. But I just realized I can remove it if I just update `functionsAndSchemas` before I check